src/main/java/org/openjdk/jextract/impl/ClassSourceBuilder.java

```diff
@@ -171,16 +180,118 @@ final void emitDocComment(Type.Function funcType, String name) {
             """);
     }
 
-    final void emitConstantGetter(String mods, String getterName, boolean nullCheck, String symbolName, Constant constant, Declaration decl) {
-        incrAlign();
-        if (decl != null) {
-            emitDocComment(decl);
+    public String mangleName(String javaName, Class<?> type) {
+        return javaName + nameSuffix(type);
+    }
+
+    String nameSuffix(Class<?> type) {
+        if (type.equals(MemorySegment.class)) {
+            return "$SEGMENT";
+        } else if (type.equals(MemoryLayout.class)) {
+            return "$LAYOUT";
+        } else if (type.equals(MethodHandle.class)) {
+            return "$MH";
+        } else if (type.equals(VarHandle.class)) {
+            return "$VH";
+        } else if (type.equals(FunctionDescriptor.class)) {
+            return "$DESC";
+        } else {
+            return "";
         }
-        appendLines(STR."""
-            \{mods} \{constant.type().getSimpleName()} \{getterName}() {
-                return \{nullCheck ? STR."RuntimeHelper.requireNonNull(\{constant}, \"\{symbolName}\")" : constant};
+    }
+
+    String layoutString(int textBoxIndent, MemoryLayout l) {
+        StringBuilder builder = new StringBuilder();
+        layoutString(textBoxIndent, builder, l);
+        return builder.toString();
+    }
+
+    private void layoutString(int textBoxIndent, StringBuilder builder, MemoryLayout l) {
```

Comment:
Note: the logic is a bit convoluted because we have to add some indentation to the resulting string, as it can span multiple lines. And so that it is correctly aligned inside the text box in which it appears into.

Reply:
Ideally, we could avoid all this by just using low-level append calls - but that would make it hard for clients to just embed the layout inside a string template, and make the code less readable, so I went for this instead.